File: src/AddToCart.java
package com.example.tests;

import java.util.regex.Pattern;
import java.util.concurrent.TimeUnit;
import org.testng.annotations.*;
import static org.testng.Assert.*;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class AddToCart {
  private WebDriver driver;
  private String baseUrl;
  private boolean acceptNextAlert = true;
  private StringBuffer verificationErrors = new StringBuffer();

  @BeforeClass(alwaysRun = true)
  public void setUp() throws Exception {
    driver = new FirefoxDriver();
    baseUrl = "https://www.newegg.ca/";
    driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
  }

  @Test
  public void testAddToCart() throws Exception {
    driver.get(baseUrl + "/Product/Product.aspx?Item=9SIA9U16WU4929");
    try {
      assertEquals(driver.findElement(By.xpath("(//button[@type='button'])[61]")).getText(), "ADD TO CART");
    } catch (Error e) {
      verificationErrors.append(e.toString());
    }
    driver.findElement(By.xpath("(//button[@type='button'])[61]")).click();
    driver.findElement(By.cssSelector("div.item-button-area > button.btn.btn-primary")).click();
    driver.findElement(By.linkText("Secure Checkout")).click();
    driver.findElement(By.id("UserName")).clear();
    driver.findElement(By.id("UserName")).sendKeys("breadpowder@gmail.com");
    driver.findElement(By.id("UserPwd")).clear();
    driver.findElement(By.id("UserPwd")).sendKeys("Qianggou1");
    driver.findElement(By.id("submit")).click();
    driver.findElement(By.id("cvv2Code")).clear();
    driver.findElement(By.id("cvv2Code")).sendKeys("1234");
    driver.findElement(By.id("term")).click();
    try {
      assertEquals(driver.findElement(By.id("SubmitOrder")).getText(), "Place Order");
    } catch (Error e) {
      verificationErrors.append(e.toString());
    }
  }

  @AfterClass(alwaysRun = true)
  public void tearDown() throws Exception {
    driver.quit();
    String verificationErrorString = verificationErrors.toString();
    if (!"".equals(verificationErrorString)) {
      fail(verificationErrorString);
    }
  }

  private boolean isElementPresent(By by) {
    try {
      driver.findElement(by);
      return true;
    } catch (NoSuchElementException e) {
      return false;
    }
  }

  private boolean isAlertPresent() {
    try {
      driver.switchTo().alert();
      return true;
    } catch (NoAlertPresentException e) {
      return false;
    }
  }

  private String closeAlertAndGetItsText() {
    try {
      Alert alert = driver.switchTo().alert();
      String alertText = alert.getText();
      if (acceptNextAlert) {
        alert.accept();
      } else {
        alert.dismiss();
      }
      return alertText;
    } finally {
      acceptNextAlert = true;
    }
  }
}

File: src/AddToCart.py
# -*- coding: utf-8 -*-
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import Select
from selenium.common.exceptions import NoSuchElementException
from selenium.common.exceptions import NoAlertPresentException
import unittest, time, re

class AddToCart(unittest.TestCase):
    def setUp(self):
        self.driver = webdriver.Firefox()
        self.driver.implicitly_wait(30)
        self.base_url = "https://www.newegg.ca/"
        self.verificationErrors = []
        self.accept_next_alert = True
    
    def test_add_to_cart(self):
        driver = self.driver
        driver.get(self.base_url + "/Product/Product.aspx?Item=9SIA9U16WU4929")
        try: self.assertEqual("ADD TO CART", driver.find_element_by_xpath("(//button[@type='button'])[61]").text)
        except AssertionError as e: self.verificationErrors.append(str(e))
        driver.find_element_by_xpath("(//button[@type='button'])[61]").click()
        driver.find_element_by_css_selector("div.item-button-area > button.btn.btn-primary").click()
        driver.find_element_by_link_text("Secure Checkout").click()
        driver.find_element_by_id("UserName").clear()
        driver.find_element_by_id("UserName").send_keys("breadpowder@gmail.com")
        driver.find_element_by_id("UserPwd").clear()
        driver.find_element_by_id("UserPwd").send_keys("Qianggou1")
        driver.find_element_by_id("submit").click()
        driver.find_element_by_id("cvv2Code").clear()
        driver.find_element_by_id("cvv2Code").send_keys("1234")
        driver.find_element_by_id("term").click()
        try: self.assertEqual("Place Order", driver.find_element_by_id("SubmitOrder").text)
        except AssertionError as e: self.verificationErrors.append(str(e))
    
    def is_element_present(self, how, what):
        try: self.driver.find_element(by=how, value=what)
        except NoSuchElementException as e: return False
        return True
    
    def is_alert_present(self):
        try: self.driver.switch_to_alert()
        except NoAlertPresentException as e: return False
        return True
    
    def close_alert_and_get_its_text(self):
        try:
            alert = self.driver.switch_to_alert()
            alert_text = alert.text
            if self.accept_next_alert:
                alert.accept()
            else:
                alert.dismiss()
            return alert_text
        finally: self.accept_next_alert = True
    
    def tearDown(self):
        self.driver.quit()
        self.assertEqual([], self.verificationErrors)

if __name__ == "__main__":
    unittest.main()
